frontend/src/component/insights/sections/PerformanceInsights.tsx

```diff
@@ -22,13 +22,42 @@ import {
     StyledWidget,
     StyledWidgetContent,
     StyledWidgetStats,
-    StatsExplanation,
 } from '../InsightsCharts.styles';
 import { useUiFlag } from 'hooks/useUiFlag';
 import { NewProductionFlagsChart } from '../componentsChart/NewProductionFlagsChart/NewProductionFlagsChart.tsx';
-import Lightbulb from '@mui/icons-material/LightbulbOutlined';
 import { CreationArchiveChart } from '../componentsChart/CreationArchiveChart/CreationArchiveChart.tsx';
 import { CreationArchiveStats } from '../componentsStat/CreationArchiveStats/CreationArchiveStats.tsx';
+import { NewProductionFlagsStats } from '../componentsStat/NewProductionFlagsStats/NewProductionFlagsStats.tsx';
+import { useProductionFlagsData } from '../componentsChart/NewProductionFlagsChart/useNewProductionFlagsData.ts';
+
+const NewProductionFlagsWidget = ({
+    groupedLifecycleData,
+    loading,
+    showAllProjects,
+    labels,
+}) => {
+    const { median, chartData } = useProductionFlagsData({
+        groupedLifecycleData,
+        labels,
+        loading,
+        showAllProjects,
+    });
+
+    return (
+        <StyledWidget>
+            <StyledWidgetStats width={275}>
+                <WidgetTitle title='New flags in production' />
+                <NewProductionFlagsStats
+                    chartData={chartData}
+                    median={median}
+                />
+            </StyledWidgetStats>
+            <StyledChartContainer>
+                <NewProductionFlagsChart chartData={chartData} />
+            </StyledChartContainer>
+        </StyledWidget>
+    );
+};
```

Comment:
This is the reason why there's so much refactoring in this PR. We previously aggregated the data within the newproductionflags component. But because we now need to share it between that component and the stat card, we need to extract it and bring it up a level in the tree.